File: exercices5/Exo6.py
class PileBorne:
    def __init__(self, tailleMax):
        self._tailleMax = tailleMax
        self.contenu = list()

    def __len__(self):
        return len(self.contenu)

    def est_vide(self):
        return self.__len__() == 0

    def est_pleine(self):
        return self.__len__() == self._tailleMax

    def empiler(self, x):
        if(self.__len__() < self._tailleMax):   
            self.contenu.append(x)
            return
        raise IndexError("Taille maximale")

    def depiler(self):
        if(not self.est_vide()):
            return self.contenu.pop(0)
        raise IndexError("Aucun element dans la pile")

File: exercices5/cpp/Exo6.cpp
#include "./Exo6.hpp"

PileBornee::PileBornee(int capacite)
{
    this->capacite = capacite;
}

bool PileBornee::est_vide()
{
    return this->nb == 0;
}

bool PileBornee::est_pleine()
{
    // 
    return this->nb == this->contenu.size();
}

void PileBornee::empiler(int x)
{
    if(this->est_pleine())
        // error
    this->contenu.push_back(x);
    this->nb ++;
}

void PileBornee::depiler()
{
    if(this->est_vide())
        // error
    //int valeur = this->contenu.;
    return this->contenu.pop_back()
}